rclpy/rclpy/executors.py

```diff
@@ -429,7 +448,8 @@ def can_execute(self, entity: WaitableEntityType) -> bool:
     def _wait_for_ready_callbacks(
         self,
         timeout_sec: float = None,
-        nodes: List['Node'] = None
+        nodes: List['Node'] = None,
+        condition: Callable[[], bool] = lambda: True,
```

Comment:
```suggestion
        condition: Callable[[], bool] = lambda: False,
```

unless the following error will happen with `SingleThreadedExecutor`

```
ros2 run my_package my_node --exe=S
[INFO] [1595381337.338499864] [pubnode_22677]: Publishing hello1: Thread MainThread
Using <class 'rclpy.executors.SingleThreadedExecutor'>
Traceback (most recent call last):
  File "/root/ros2_ws/tmp_ws/install/my_package/lib/my_package/my_node", line 11, in <module>
    load_entry_point('my-package', 'console_scripts', 'my_node')()
  File "/root/ros2_ws/tmp_ws/build/my_package/my_package/my_node.py", line 70, in main
    exe.spin_until_future_complete(future_msgs)
  File "/root/ros2_ws/tmp_ws/install/rclpy/lib/python3.8/site-packages/rclpy/executors.py", line 296, in spin_until_future_complete
    self.spin_once_until_future_complete(future, timeout_sec)
  File "/root/ros2_ws/tmp_ws/install/rclpy/lib/python3.8/site-packages/rclpy/executors.py", line 713, in spin_once_until_future_complete
    self.spin_once(timeout_sec)
  File "/root/ros2_ws/tmp_ws/install/rclpy/lib/python3.8/site-packages/rclpy/executors.py", line 702, in spin_once
    handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
  File "/root/ros2_ws/tmp_ws/install/rclpy/lib/python3.8/site-packages/rclpy/executors.py", line 688, in wait_for_ready_callbacks
    return next(self._cb_iter)
  File "/root/ros2_ws/tmp_ws/install/rclpy/lib/python3.8/site-packages/rclpy/executors.py", line 668, in _wait_for_ready_callbacks
    raise ConditionReachedException()
rclpy.executors.ConditionReachedException
```

Reply:
Thanks!
I inverted the logic before committing, and I didn't realize that I forgot to change the `condition` argument default.